### ISSUE TITLE
AttributeError: 'SocketPort' object has no attribute '_lock'

### ISSUE THREAD

**Issue by retroj** [olemb/mido#79]
Hello, I receive an error when trying to connect to a tcp socketport.

```
gld_address = ("192.168.2.49", 51325)
gld_connection = SocketPort(*gld_address)

```
```
Traceback (most recent call last):
  File "__init__.py", line 20, in <module>
    for message in gld_connection:
  File "/usr/lib/python3.6/site-packages/mido/ports.py", line 238, in __iter__
    yield self.receive()
  File "/usr/lib/python3.6/site-packages/mido/ports.py", line 187, in receive
    with self._lock:
AttributeError: 'SocketPort' object has no attribute '_lock'
Exception ignored in: <bound method BasePort.__del__ of <open I/O '192.168.2.4951325' (SocketPort)>>
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/mido/ports.py", line 81, in __del__
  File "/usr/lib/python3.6/site-packages/mido/ports.py", line 69, in close
AttributeError: 'SocketPort' object has no attribute '_lock'

```
I can work around the error by modifying the program as follows:

```
gld_connection = SocketPort(*gld_address)
gld_connection.__dict__['_lock'] = threading.RLock() ## sick hack

```
I believe that the problem is that SocketPort does not call its super's __init__ method.

**Comment by olemb** [comment]
This was fixed by in request #72 but I haven't had time to release it yet. I hope to get a new version out tonight.

Thanks for reporting it!

**Comment by olemb** [comment]
1.1.19 is out. Socket ports should work now.
